src/backtrack/TraverseMaze.java

```diff
@@ -0,0 +1,108 @@
+package backtrack;
+
+import java.util.*;
+
+public class TraverseMaze {
+
+    private int[][] moveUp = {{1, 0}};
+    private int[][] moveDown = {{-1, 0}};
+    private int[][] moveLeft = {{0, -1}};
+    private int[][] moveRight = {{0, 1}};
+    private String[][] maze;
+    private Set<int[][]> alreadyVisited = new HashSet<>();
+    private Deque<int[][]> thePathStack = new ArrayDeque<>();
+
+    public TraverseMaze(String[][] traverseMaze) {
+        this.maze = traverseMaze;
+    }
+
+    public int[][] moveLeft(int[][] currentPosition) {
+        int newPositionX = currentPosition[0][0] + moveLeft[0][0];
+        int newPositionY = currentPosition[0][1] + moveLeft[0][1];
+        int[][] newCurrentPosition = {{newPositionX, newPositionY}};
+        System.out.println();
+        return newCurrentPosition;
+    }
+
+    public int[][] moveRight(int[][] currentPosition) {
+        int newPositionX = currentPosition[0][0] + moveRight[0][0];
+        int newPositionY = currentPosition[0][1] + moveRight[0][1];
+        int[][] newCurrentPosition = {{newPositionX, newPositionY}};
+        System.out.println();
+        return newCurrentPosition;
+    }
+
+    public int[][] moveDown(int[][] currentPosition) {
+        int newPositionX = currentPosition[0][0] + moveDown[0][0];
+        int newPositionY = currentPosition[0][1] + moveDown[0][1];
+        int[][] newCurrentPosition = {{newPositionX, newPositionY}};
+        System.out.println();
+        return newCurrentPosition;
+
+    }
+
+    public int[][] moveUp(int[][] currentPosition) {
+        int newPositionX = currentPosition[0][0] + moveUp[0][0];
+        int newPositionY = currentPosition[0][1] + moveUp[0][1];
+        int[][] newCurrentPosition = {{newPositionX, newPositionY}};
+        return newCurrentPosition;
+
+    }
+
+    public String move(int[][] initialPosition) {
```

Comment:
I think this method should be renamed to solveMaze or findPath...
It is not about just moving but finding a solution
Maybe it should return just a boolean, no need to a string IMO